ui_framework/src/components/code_editor/code_editor.js

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import AceEditor from 'react-ace';
+
+import { htmlIdGenerator, keyCodes } from '../../../services';
+
+export class KuiCodeEditor extends Component {
+
+  state = {
+    isHintActive: true
+  };
+
+  idGenerator = htmlIdGenerator();
+
+  aceEditorRef = (aceEditor) => {
+    if (aceEditor) {
+      this.aceEditor = aceEditor;
+      aceEditor.editor.textInput.getElement().tabIndex = -1;
+      aceEditor.editor.textInput.getElement().addEventListener('keydown', this.onKeydownAce);
+    }
+  };
+
+  stopEditing() {
```

Comment:
I'd still move this down so it's after `startEditing`.

Reply:
Done